src/wxflow/task.py

```diff
@@ -60,6 +62,9 @@ def __init__(self, config: Dict, *args, **kwargs):
         self.runtime_config['previous_cycle'] = add_to_datetime(self.runtime_config.current_cycle, -to_timedelta(f"{self.config['assim_freq']}H"))
         logger.debug(f"previous cycle: {self.runtime_config['previous_cycle']}")
 
+        # Combine config and runtime_config into single task_config attribute-dictionary
+        self.task_config = AttrDict(**self.config, **self.runtime_config)
```

Comment:
This will not force users to strictly use only `self.task_config`
I think we should make `config` and `runtime_config` internal (private) and only return `self.task_config`.

Change:
L35:
```python
self._config = AttrDict(config)
```
and L44
```python
self._runtime_config = AttrDict()
```
and then update the script to use these.

Reply:
Ok, good idea. I will have to make a companion PR for the Global Workflow though since config and runtime_config are referenced in several places.

Reply:
Yes. Unfortunately, it wasn't caught in the global-workflow at the time this happened, but this is necessary before we promote too many kinds of `_config`'s.